packages/desktop-client/src/components/modals/manager/DuplicateFileModal.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+
+import { addNotification, duplicateBudget } from 'loot-core/client/actions';
+import { type File } from 'loot-core/src/types/file';
+
+import { theme } from '../../../style';
+import { Button, ButtonWithLoading } from '../../common/Button2';
+import { FormError } from '../../common/FormError';
+import { InitialFocus } from '../../common/InitialFocus';
+import { InlineField } from '../../common/InlineField';
+import { Input } from '../../common/Input';
+import {
+  Modal,
+  ModalButtons,
+  ModalCloseButton,
+  ModalHeader,
+} from '../../common/Modal';
+import { Text } from '../../common/Text';
+import { View } from '../../common/View';
+
+type DuplicateFileProps = {
+  file: File;
+  managePage?: boolean;
+  loadBudget?: 'none' | 'original' | 'copy';
+  onComplete?: (event: {
+    status: 'success' | 'failed' | 'canceled';
+    error?: object;
+  }) => void;
+};
+
+export function DuplicateFileModal({
+  file,
+  managePage,
+  loadBudget = 'none',
+  onComplete,
+}: DuplicateFileProps) {
+  const { t } = useTranslation();
+  const [newName, setNewName] = useState(file.name + ' - copy');
+  const [nameError, setNameError] = useState<string | null>(null);
+
+  // If the state is "broken" that means it was created by another user.
+  const isCloudFile = 'cloudFileId' in file && file.state !== 'broken';
+  const dispatch = useDispatch();
+
+  const [loadingState, setLoadingState] = useState<'cloud' | 'local' | null>(
+    null,
+  );
+
+  const validateNewName = (name: string): string | null => {
+    const trimmedName = name.trim();
+    if (trimmedName === '') return t('Budget name cannot be blank');
+    if (trimmedName.length > 100) {
+      return t('Budget name is too long (max length 100)');
+    }
+    if (!/^[a-zA-Z0-9 .\-_()]+$/.test(trimmedName)) {
+      return t('Budget name contains invalid characters');
+    }
+    // Additional validation checks can go here
```

Comment:
NIT: Seems like a needless comment

Reply:
Ha! That's what I get for not thoroughly reviewing coderabbitai's recommendations.
I'll remove the comment.

Reply:
Fixed in commit [89a9907](https://github.com/actualbudget/actual/pull/3847/commits/89a9907082112209d66c3986046dd7d54d030a41)